### PR TITLE
Add rack attack for SPAM control

### PR DESCRIPTION
**Pull request by gauravtiwari**
@jgorset What you think? Also, do we have a list of bot urls or something that we can use to blacklist them.

### CODE REVIEW

**Comment by gauravtiwari** [comment]
@jgorset Think we can merge this

File: config/initializers/rack_attack.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  # Blacklist spammers
+  spammer_regexp = Regexp.union(spammers)
+  blacklist("block referer spam") do |request|
+    request.referer =~ spammer_regexp
+  end
+
+  # Throttle room creation
+  throttle('new/rooms', :limit => 2, :period => 24.hours) do |req|
+    req.post? && req.path == "/rooms"
+  end
```
Comment (jgorset) [inline config/initializers/rack_attack.rb]: I think we might have some issues with this; for example, if you try to submit a room with validation errors twice. It's also feasible that we might create more than two rooms per day. The honeypot alone should do it, though, right?

**Comment by jgorset** [comment]
Nice @gauravtiwari! I left a comment about the rate limiting that I'd appreciate your input on, but other than that I second that we're good to merge!

File: config/initializers/rack_attack.rb (hunk 1 above)
Comment (gauravtiwari) [inline config/initializers/rack_attack.rb]: @jgorset: Correct :+1:! Should we remove it then or increase the numbers, perhaps 5 or 10 then?

Comment (jgorset) [inline config/initializers/rack_attack.rb]: I think removing it makes the most sense, then – it's only likely to trigger for humans.

**Comment by gauravtiwari** [comment]
@jgorset All good to go :)

**Comment by gauravtiwari** [comment]
**Ping Pong**

File: config/initializers/rack_attack.rb [hunk 2]
```diff
@@ -12,8 +12,8 @@ class Rack::Attack
...
   end
 
-  # Throttle room creation
-  throttle('new/rooms', :limit => 2, :period => 24.hours) do |req|
-    req.post? && req.path == "/rooms"
-  end
+  # # Throttle room creation
+  # throttle('new/rooms', :limit => 2, :period => 24.hours) do |req|
+  #   req.post? && req.path == "/rooms"
+  # end
```
Comment (jgorset) [inline config/initializers/rack_attack.rb]: No need to comment out code; that's what version control is for, after all. We can just remove it and revert the commit later if we want it back.

**Pull request closed without merging**